sql/expression_value.h

```diff
@@ -1526,6 +1530,60 @@ struct UnknownRowValueInfo: public RowValueInfo {
     }
 };
 
+/*****************************************************************************/
+/* OR Expression Value Info                                                  */
+/*****************************************************************************/
+
+/** Expression Value info when we dont know which of two value info we will get 
+    With a Case for Example.
+*/
+
+struct ORExpressionValueInfo: public ExpressionValueInfoT<ExpressionValue> {
+
+    ORExpressionValueInfo(std::shared_ptr<ExpressionValueInfo> left, std::shared_ptr<ExpressionValueInfo> right) {
+        left_ = left;
+        right_ = right;
+    }
+
+    virtual bool isScalar() const { return left_->isScalar() && right_->isScalar(); }
+
+    virtual std::shared_ptr<RowValueInfo> getFlattenedInfo() const;
+
+    virtual void flatten(const ExpressionValue & value,
+                         const std::function<void (const ColumnName & columnName,
+                                                   const CellValue & value,
+                                                   Date timestamp)> & write)
+        const;
+
+    virtual bool isCompatible(const ExpressionValue & value) const
+    {
+        return left_->isCompatible(value) && right_->isCompatible(value);
+    }
+
+    virtual SchemaCompleteness getSchemaCompleteness() const {
+        return std::min(left_->getSchemaCompleteness(), right_->getSchemaCompleteness());
```

Comment:
This is relying on the order of the enum.  I think 
```
return left_->getSchemaCompleteness() == SCHEMA_CLOSED && right_->getSchemaCompleteness() == SCHEMA_CLOSED ? SCHEMA_CLOSED : SCHEMA_OPEN;
```
is more explicit and less error prone.